modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/authenticator/LDAPAuth.java

```diff
@@ -512,9 +523,19 @@ private int _authenticate(
 		List<LDAPServerConfiguration> ldapServerConfigurations =
 			_ldapServerConfigurationProvider.getConfigurations(companyId);
 
+		boolean userImportedByLDAP = false;
+
 		for (LDAPServerConfiguration ldapServerConfiguration :
 				ldapServerConfigurations) {
 
+			Binding ldapUser = _portalLDAP.getUser(
+				ldapServerConfiguration.ldapServerId(), companyId, screenName,
+				emailAddress);
+
+			if (ldapUser != null) {
+				userImportedByLDAP = true;
+			}
+
```

Comment:
Same question as below. I couldn't figure out why is this required.

Reply:
This block checks if the user was imported by LDAP, so that automatic login does not happen.